core/src/appleMain/kotlin/Peripheral.kt

```diff
@@ -224,6 +203,9 @@ public class ApplePeripheral internal constructor(
     }
 
     public override suspend fun connect() {
+        //need to check that coreblueooth is turned on and available
+        centralManager.delegate.state
+            .first { state -> state == CBCentralManagerStatePoweredOn }
```

Comment:
fair enough.  I have changed the PR to reflect this.  In order to get the bluetooth status, I altered the bluetooth state flow and set it as a StateFlow instead of a Flow.  This was the easiest thing I could think of to keep it from suspending.